### DISCUSSION TITLE
Springboot - Gradle task for buildpacks new-relic

### DISCUSSION THREAD

**Discussion by kenwiths** [paketo-buildpacks/feedback#177] [Q&A] [answered]
I want to import new-relic into my spring-boot-app image without to use "COPY new-relic.jar" in Dockerfile for runImage. I tried this task in build.gradle:
```java
bootBuildImage {
    builder = "paketobuildpacks/builder:tiny"
    buildpacks = [
            "gcr.io/paketo-buildpacks/ca-certificates",
            "gcr.io/paketo-buildpacks/amazon-corretto",
            "gcr.io/paketo-buildpacks/syft",
            "gcr.io/paketo-buildpacks/executable-jar",
            "gcr.io/paketo-buildpacks/dist-zip",
            "gcr.io/paketo-buildpacks/spring-boot",
            "gcr.io/paketo-buildpacks/new-relic"
    ]
    environment = [
            "BP_JVM_VERSION" : "17.*",
            "BPL_SPRING_CLOUD_BINDINGS_DISABLED": "true",
            "JAVA_TOOL_OPTIONS": "-XX:ActiveProcessorCount=4"
    ]
    runImage = "paketobuildpacks/run:tiny"
}
``` 
Then run with this command:
`./gradlew bootBuildImage -p apilayer/ --imageName="test"`

And the **Error** return below:
```
> Running creator
    [creator]     ===> ANALYZING
    [creator]     ===> DETECTING
    [creator]     ======== Results ========
    [creator]     pass: paketo-buildpacks/ca-certificates@3.6.0
    [creator]     pass: paketo-buildpacks/amazon-corretto@8.0.0
    [creator]     pass: paketo-buildpacks/syft@1.26.0
    [creator]     pass: paketo-buildpacks/executable-jar@6.6.2
    [creator]     pass: paketo-buildpacks/dist-zip@5.5.2
    [creator]     pass: paketo-buildpacks/spring-boot@5.23.0
    [creator]     fail: paketo-buildpacks/new-relic@5.7.1
    [creator]     ERROR: No buildpack groups passed detection.
    [creator]     ERROR: Please check that you are running against the correct path.
    [creator]     ERROR: failed to detect: no buildpacks participating

> Task :apilayer:bootBuildImage FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':apilayer:bootBuildImage'.
> Builder lifecycle 'creator' failed with status code 20

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 22s
15 actionable tasks: 5 executed, 10 up-to-date
``` 
I think `paketo-buildpacks/new-relic` not incompatible with my app. But I still want to ask u guys this solution is good ? or any ideas for this ?

> Spring boot 2.7.9
Java 17 corretto
Gradle 7.4

**Comment by dmikusa** [comment, marked as answer]
Short answer: you need to include a binding for the New Relic buildpack to detect properly.

>A binding exists with type of NewRelic

https://github.com/paketo-buildpacks/new-relic/#behavior

See [this doc page](https://paketo.io/docs/howto/configuration/#bindings) for details on bindings. In a nutshell, a binding is just a folder with files. One file called `type` with a value of `NewRelic` and then additional files for settings you want to pass to the New Relic buildpack, where the name is the key and the contents are the value. For example, a file called `license_key` with the contents of `<your license>` would get passed through to New Relic as it's license key (it's passed through via an env variable `NEW_RELIC_LICENSE_KEY` which is automatically set).

You then need to set the binding as a volume mounted to the app at build time and at runtime. At build time, you can do that by adding `bindings = ['${projectDir}/bindings/new-relic:/platform/bindings/new-relic']` to the `bootBuildImage` section in your `build.gradle` file. At runtime, it depends on how you're running the container image but with docker it's the `--volume` argument for `docker run`.
